include/seqan3/alignment/pairwise_alignment_edit_distance_unbanded.hpp

```diff
@@ -0,0 +1,548 @@
+    }
+
+    //!\brief Decrement the last active cell position.
+    inline bool prev_last_active_cell()
+    {
+        score_mask >>= 1;
+        if (score_mask != (word_type)0)
+            return true;
+
+        last_block--;
+        if (is_global && last_block == (unsigned)-1)
+            return false;
+
+        score_mask = (word_type)1 << (word_size - 1);
+        return true;
+    }
+
+    //!\brief Increment the last active cell position.
+    inline void next_last_active_cell()
+    {
+        score_mask <<= 1;
+        if (score_mask)
+            return;
+
+        score_mask = 1;
+        last_block++;
+    }
+
+    //!\brief Use the ukkonen trick and update the last active cell.
+    inline bool update_last_active_cell()
+    {
+        // updating the last active cell
+        while (!(_score <= max_errors))
+        {
+            advance_score(vp[last_block], vn[last_block], score_mask);
+            if(!prev_last_active_cell())
+                break;
+        }
+
+        if ((score_mask == last_score_mask) && (last_block == vp.size() - 1))
+            return on_hit();
+        else
+        {
+            next_last_active_cell();
+            advance_score(vp[last_block], vn[last_block], score_mask);
+        }
+
+        return false;
+    }
+
+    //!\brief Will be called if a hit was found (e.g., score < max_errors).
+    inline bool on_hit()
+    {
+        // _setFinderEnd(finder);
+        //
+        // if constexpr(is_global)
+        //     _setFinderLength(finder, endPosition());
+
+        return false;
+    }
+
+    //!\brief Pattern is small enough that it fits into one machine word. Use
+    //!faster computation with less overhead.
+    bool small_patterns();
+
+    //!\brief Pattern is larger than one machine word. Use overflow aware computation.
+    bool large_patterns();
+
+public:
+    //!\brief Compute the alignment.
+    pairwise_alignment_edit_distance_unbanded& run()
+    {
+        // limit search width for prefix search
+        if constexpr(use_max_errors && is_global)
+        {
+            std::size_t max_length = query.size() + max_errors + 1;
+            std::size_t haystack_length = std::min(database.size(), max_length);
+            database_it_end += database.size() - haystack_length;
+        }
+
+        // distinguish between the version for needles not longer than
+        // one machine word and the version for longer needles
+        if (vp.size() <= 1)
+            small_patterns();
+        else
+            large_patterns();
+
+        return *this;
+    }
+
+    //!\brief Return the score of the alignment.
+    int score()
+    {
+        return _score;
+    }
+
+    //!\brief Return the score matrix of the alignment.
+    score_matrix_type score_matrix()
+    {
+        return score_matrix_type{*this};
+    }
+
+    //!\brief Return the trace matrix of the alignment.
+    trace_matrix_type trace_matrix()
+    {
+        return trace_matrix_type{*this};
+    }
+};
```

Comment:
there are 6(!) switches between public and private sections in this class, please clean this up a little, it's very hard to follow

Reply:
and put members into groups!

Reply:
I wanted to put them into groups, but this does not work apparently when mixing type definitions, functions and variables.